exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/RestrictingNodeDecorator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.core.runtime;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.core.service.Node;
+import com.exonum.binding.core.storage.database.Snapshot;
+import com.exonum.binding.core.transaction.RawTransaction;
+import java.util.function.Function;
+
+/**
+ * Node decorator which can restrict access to the node by calling {@link #restrictAccess()} method.
+ */
+class RestrictingNodeDecorator implements Node {
```

Comment:
I'd ponder on a name that communicates that this decorator does not close the Node
it decorates:
 - NonOwningNodeDecorator (it does not _own_ the decorated Node, hence, does not destroy it)?
 - NonDestroyingNodeDecorator?
 - MultiplexingNodeDecorator?